### PR TITLE
Login view

### PR DESCRIPTION
**Pull request by Regit68**

### CODE REVIEW

File: Navigation/Profile/LogInViewController.swift [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+            containerView.widthAnchor.constraint(equalTo: scrollView.widthAnchor),
+            
+            logoVK.topAnchor.constraint(equalTo: containerView.topAnchor, constant: 120),
+            logoVK.centerXAnchor.constraint(equalTo: containerView.centerXAnchor),
+            logoVK.widthAnchor.constraint(equalToConstant: logoVKWidth),
+            logoVK.heightAnchor.constraint(equalToConstant: logoVKWidth),
+            
+            emailPhoneTextField.topAnchor.constraint(equalTo: logoVK.bottomAnchor, constant: 120),
+            emailPhoneTextField.leadingAnchor.constraint(equalTo: containerView.safeAreaLayoutGuide.leadingAnchor, constant: 16),
+            emailPhoneTextField.trailingAnchor.constraint(equalTo: containerView.safeAreaLayoutGuide.trailingAnchor, constant: -16),
```
Comment (Alexdive) [inline Navigation/Profile/LogInViewController.swift]: No need to use safeAreaLayoutGuide here and after.
Logically you would want to use on the first container constraints, which is scrollView in this case 
Other containers will use bounds of parent

File: Navigation/Profile/PostTableViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+//
+//  Created by Irina on 07.02.2022.
+//
+
+import UIKit
+
+class PostTableViewCell: UITableViewCell {
+    var post: Post? {
+        didSet {
+            guard let post = post else { return }
```
Comment (Alexdive) [inline Navigation/Profile/PostTableViewCell.swift:13]: Nice!

File: Navigation/Profile/PostTableViewCell.swift [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        label.font = UIFont.systemFont(ofSize: 20, weight: .bold)
+        label.textColor = .black
+        label.numberOfLines = 2
+        return label
+    }()
+    
+    private let postImageView: UIImageView = {
+        let imageView = UIImageView()
+        /// Adapts to screen size.
+        imageView.contentMode = .scaleAspectFit
```
Comment (Alexdive) [inline Navigation/Profile/PostTableViewCell.swift:39]: Actually it adapts image to postImageView size

File: Navigation/Profile/PostTableViewCell.swift [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+            likesLabel.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: 16),
+            likesLabel.bottomAnchor.constraint(equalTo: contentView.bottomAnchor, constant: -16),
+            
+            viewsLabel.topAnchor.constraint(equalTo: descriptionLabel.bottomAnchor, constant: 10),
+            viewsLabel.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -16),
+            viewsLabel.bottomAnchor.constraint(equalTo: contentView.bottomAnchor, constant: -16)
+        ])
+    }
+}
+
```
Comment (Alexdive) [inline Navigation/Profile/PostTableViewCell.swift]: Remove all unneeded code and commented code below.
File should end with one empty line after last }

File: Navigation/Profile/ProfileHeaderView.swift [hunk 5]
```diff
@@ -124,7 +124,8 @@ class ProfileHeaderView: UIView {
             setStatusButton.topAnchor.constraint(equalTo: statusTextField.bottomAnchor, constant: 10),
             setStatusButton.leadingAnchor.constraint(equalTo: safeAreaLayoutGuide.leadingAnchor, constant: 16),
             setStatusButton.trailingAnchor.constraint(equalTo: safeAreaLayoutGuide.trailingAnchor, constant: -16),
```
Comment (Alexdive) [inline Navigation/Profile/ProfileHeaderView.swift]: same for safeArea

File: Navigation/Profile/ProfileViewController.swift [hunk 6]
```diff
@@ -7,48 +7,145 @@
 
 import UIKit
 
-class ProfileViewController: UIViewController {
+class ProfileViewContoller: UIViewController {
+    
+    private let posts: [String] = [
```
Comment (Alexdive) [inline Navigation/Profile/ProfileViewController.swift]: Why do you need this?

File: Navigation/Profile/ProfileViewController.swift [hunk 7]
```diff
@@ -7,48 +7,145 @@
 
 import UIKit
 
-class ProfileViewController: UIViewController {
+class ProfileViewContoller: UIViewController {
```
Comment (Alexdive) [inline Navigation/Profile/ProfileViewController.swift]: Typo in class name

File: Navigation/Profile/ProfileViewController.swift [hunk 8]
```diff
@@ -7,48 +7,145 @@
...
         
-        navigationController?.navigationBar.prefersLargeTitles = false
-        navigationItem.title = "Profile"
+        NSLayoutConstraint.activate([
+            tableView.topAnchor.constraint(equalTo: view.topAnchor),
+            tableView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            tableView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            tableView.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+            
+            profileHeaderView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
```
Comment (Alexdive) [inline Navigation/Profile/ProfileViewController.swift]: You don't have profileHeaderView added to hierarchy, so these constraints are useless

File: Navigation/Profile/ProfileViewController.swift [hunk 9]
```diff
@@ -7,48 +7,145 @@
...
         
-        view.addSubview(profileHeaderView)
-        view.addSubview(button)
+        return UITableViewCell()
+    }
+// MARK: Header for section.
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        let profileHeaderView = ProfileHeaderView()
+
+        return profileHeaderView
```
Comment (Alexdive) [inline Navigation/Profile/ProfileViewController.swift]: I would make a check for number of section, to make sure this header added to the first one

return section == 0 ? profileHeaderView : nil

File: Navigation/Profile/ProfileViewController.swift [hunk 10]
```diff
@@ -7,48 +7,145 @@
...
+            return cell
+        }
         
-        view.addSubview(profileHeaderView)
-        view.addSubview(button)
+        return UITableViewCell()
+    }
+// MARK: Header for section.
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        let profileHeaderView = ProfileHeaderView()
```
Comment (Alexdive) [inline Navigation/Profile/ProfileViewController.swift]: You have profileHeaderView instance already on line 19

File: Navigation/Profile/ProfileViewController.swift [hunk 11]
```diff
@@ -7,48 +7,145 @@
...
+        tableView.deselectRow(at: indexPath, animated: true)
+    }
+    
+    func tableView(_ tableView: UITableView, estimatedHeightForHeaderInSection section: Int) -> CGFloat {
+        return 300
     }
 }
 
-// Shows ProfileHeaderView on the ProfileViewController.
+
```
Comment (Alexdive) [inline Navigation/Profile/ProfileViewController.swift]: remove unneeded and commented code

File: Navigation/Profile/LogInViewController.swift (hunk 1 above)
Comment (Regit68) [reply Navigation/Profile/LogInViewController.swift]: Done

**Review by Alexdive — APPROVED** [review]
Good job!
Take a look at small issues I commented.

one more:
extension UITextField {

These kind of extensions better to keep in separate file in Extensions folder 
Naming convention for these file can be UITextField+Padding, so you can easily understand what is inside

File: Navigation/Profile/PostTableViewCell.swift (hunk 3 above)
Comment (Regit68) [reply Navigation/Profile/PostTableViewCell.swift:39]: Thanks!

File: Navigation/Profile/PostTableViewCell.swift (hunk 4 above)
Comment (Regit68) [reply Navigation/Profile/PostTableViewCell.swift]: Done.

File: Navigation/Profile/ProfileHeaderView.swift (hunk 5 above)
Comment (Regit68) [reply Navigation/Profile/ProfileHeaderView.swift]: Done.

File: Navigation/Profile/ProfileViewController.swift (hunk 6 above)
Comment (Regit68) [reply Navigation/Profile/ProfileViewController.swift:12]: Don't need.

Comment (Regit68) [reply Navigation/Profile/ProfileViewController.swift:12]: Done

Comment (Regit68) [reply Navigation/Profile/ProfileViewController.swift]: Don't need.
Done.

File: Navigation/Profile/ProfileViewController.swift (hunk 7 above)
Comment (Regit68) [reply Navigation/Profile/ProfileViewController.swift]: Thanks!

File: Navigation/Profile/ProfileViewController.swift (hunk 8 above)
Comment (Regit68) [reply Navigation/Profile/ProfileViewController.swift]: Done.

File: Navigation/Profile/ProfileViewController.swift (hunk 9 above)
Comment (Regit68) [reply Navigation/Profile/ProfileViewController.swift]: Thanks!

File: Navigation/Profile/ProfileViewController.swift (hunk 10 above)
Comment (Regit68) [reply Navigation/Profile/ProfileViewController.swift]: Done.

File: Navigation/Profile/ProfileViewController.swift (hunk 11 above)
Comment (Regit68) [reply Navigation/Profile/ProfileViewController.swift]: Done.

**Review by Alexdive — APPROVED** [review]

**Pull request merged**